### PR TITLE
feat: template support indentation testing

### PR DESCRIPTION
**Pull request by sadeghbarati**
sry for spam

### CODE REVIEW

**Comment by sadeghbarati** [comment]
cc @Aslemammad

**Comment by sadeghbarati** [comment]
the changes are just for test idk if these changes are right 😁

File: packages/cli/index.ts [hunk 1]
```diff
@@ -115,14 +115,14 @@ const main = defineCommand({
 
           for (const p of paths) {
             const pJsonPath = path.resolve(p, "package.json");
-            const { name } = await importPackageJson(pJsonPath);
+            const { name } = await readPackageJSON();
```
Comment (Aslemammad) [inline packages/cli/index.ts:118]: Since we have multiple paths for multiple projects, there should be a way to pass the pJsonPath to this function!

File: packages/cli/index.ts [hunk 2]
```diff
@@ -115,14 +115,14 @@ const main = defineCommand({
...
 
             if (compact) {
-              await verifyCompactMode(name);
+              await verifyCompactMode(name!);
```
Comment (Aslemammad) [inline packages/cli/index.ts:121]: verify that the name always exists! We need it :)

File: packages/cli/index.ts [hunk 3]
```diff
@@ -243,24 +242,16 @@ const main = defineCommand({
...
-  const pJson = await importPackageJson(pJsonPath);
+  const pJson = await readPackageJSON();
+  
   hijackDeps(deps, pJson.dependencies);
   hijackDeps(deps, pJson.devDependencies);
-  await fs.writeFile(pJsonPath, JSON.stringify(pJson));
+  await writePackageJSON(pJsonPath, pJson);
 
-  return () => fs.writeFile(pJsonPath, content);
+  return () => writePackageJSON(pJsonPath, pJson);
```
Comment (Aslemammad) [inline packages/cli/index.ts:254]: This is clean up function so when the command finishes running, we revert the changes, that's why we store `content`

**Review by Aslemammad — CHANGES_REQUESTED** [review]

File: packages/cli/index.ts (hunk 3 above)
Comment (sadeghbarati) [reply packages/cli/index.ts:254]: [`writePackageJSON`](https://github.com/unjs/pkg-types/blob/608aa5c3df8060e78a4632943bda138e30d4e7e5/src/index.ts#L85) is doing same thing
